mypy/main.py

```diff
@@ -205,6 +206,36 @@ def invert_flag_name(flag: str) -> str:
     return '--no-{}'.format(flag[2:])
 
 
+if sys.platform == 'win32':
+    python_executable_prefix = 'py -{}'
+else:
+    python_executable_prefix = 'python{}'
+
+
+def _python_version_from_executable(python_executable: str) -> Tuple[int, int]:
+    try:
+        check = subprocess.check_output(python_executable.split(' ') +
+                                        ['-c', 'import sys; print(repr(sys.version_info[:2]))'],
+                                        stderr=subprocess.STDOUT).decode()
+    except subprocess.CalledProcessError:
+        return sys.version_info[:2]
+    else:
+        ver = re.fullmatch(r'(\d)\.(\d+)\s*', check)
+        if ver:
+            return int(ver.group(1)), int(ver.group(2))
+        else:
+            return sys.version_info[:2]
+
+
+def _python_executable_from_version(python_version: Tuple[int, int]) -> Optional[str]:
+    str_ver = '.'.join(map(str, python_version))
+    python_ver = _python_version_from_executable(python_executable_prefix.format(str_ver))
```

Comment:
Maybe an approach that does `python_launcher(version) + ['-c', 'import sys; print(sys.executable)']` would be more better here?

In particular. that would get the full path, which the user might care about checking.

Reply:
Great minds think alike :)